File: tests/07_phases_ompss.dg/fortran/success_task_18.f90
! <testinfo>
! test_generator=config/mercurium-ompss
! </testinfo>

program main
    implicit none
    interface
        !$omp task inout(a)
        subroutine sub(a)
                implicit none
                logical :: a
        end subroutine sub
    end interface

    logical :: b
    b = .false.
    call sub(b)

    !$OMP TASKWAIT

    IF (.NOT. b) STOP 1
end program main

subroutine sub(a)
        implicit none
        logical :: a

        a = .true.
end subroutine sub

File: tests/07_phases_ompss.dg/cxx/success_task_18.cpp
/*
<testinfo>
test_generator=config/mercurium-ompss
</testinfo>
*/
template < typename T>
struct C
{

    void foo()
    {
        #pragma omp task inout(p)
        {
            p = 0;
        }
    }

    T *p;
};


int main()
{
    C<int> c;
    c.foo();

#pragma omp taskwait
}
